Reader: Add styles for Discover's sidenotes

shaunandrews opened the pull request:
Moving Discover’s `.sidenote` elements into the margins on screens that have room:

Before:
![screen shot 2016-01-24 at 11 30 36 pm](https://cloud.githubusercontent.com/assets/191598/12542533/845a2096-c2f2-11e5-87ba-90e000ae757d.png)

After:
![screen shot 2016-01-24 at 11 30 48 pm](https://cloud.githubusercontent.com/assets/191598/12542536/8c76ac68-c2f2-11e5-9e2f-848267a5b6a4.png)

On smaller screens (but not mobile) the sidenotes "float" and cause the text to wrap:
![screen shot 2016-01-24 at 11 31 49 pm](https://cloud.githubusercontent.com/assets/191598/12542539/a4f31da8-c2f2-11e5-9308-1225ee848075.png)

On mobile screens, the sidenote shows inline, but with a background color to help separate it from the rest of the content:

![screen shot 2016-01-24 at 11 32 13 pm](https://cloud.githubusercontent.com/assets/191598/12542546/c04459dc-c2f2-11e5-80cb-2739dd85b828.png)

/cc @kjellr

kjellr commented:
@shaunandrews This looks great! The only bug I've found is that I think the `>1040px` breakpoint needs a little bit of margin too: I'm seeing some tight margins + cut-off content between 1090px and 1041px.

(Mac, Chrome)
<img width="1049" alt="screen shot 2016-01-25 at 10 19 38 am" src="https://cloud.githubusercontent.com/assets/1202812/12555047/42c6874a-c34d-11e5-98a6-48ab0f4ddd64.png">
<img width="1049" alt="screen shot 2016-01-25 at 10 19 49 am" src="https://cloud.githubusercontent.com/assets/1202812/12555046/42b137be-c34d-11e5-85a4-f9deeb4e6007.png">

shaunandrews commented:
I added another breakpoint (outside of our predefined ones in the breakpoint mixin) that does a kind of mix of the positioning and floating:

![screen shot 2016-01-25 at 2 00 27 pm](https://cloud.githubusercontent.com/assets/191598/12561054/164a65aa-c36c-11e5-849e-660b3559e14f.png)

I also simplified the styling of blockquotes (no background or border, larger font-size), along with images and other elements that use the `.align{left\right\center)` (no extra borders) classes:

![slice 1](https://cloud.githubusercontent.com/assets/191598/12561154/9e3b6dba-c36c-11e5-8a8f-1954af98fd1c.png)

kjellr commented:
:+1: Looks great!

blowery commented:
new blockquote styles look soooo good
